stretch_core/stretch_core/joint_trajectory_server.py

```diff
@@ -310,6 +314,10 @@ def execute_cb(self, goal_handle):
                 self.feedback_callback(goal_handle, start_time=ts)
                 # self.action_server_rate.sleep()
 
+            # TODO: We should change time.sleep to a rate.sleep(), so control
```

Comment:
Is there a reason to not add this now? Should be a small change, right?

Reply:
It's a small change but would require thorough testing. Because currently, the execution of the action happens entirely as one "unit of work" (e.g., it uses a whole thread, and doesn't hand control back to the executor until the action execution is done). IMO it is better (and more common in ROS2) to hand control back to the executor to execute other work while sleeping. But that could introduce other issues (like not having enough threads to take back control from the executor). So I think it's better for this to be in a separate PR for us to have a dedicated focus on testing this change in isolation.

Reply:
Okay, sounds good. Please open a Github issue to add this as a TODO.

Reply:
Done. https://github.com/hello-robot/stretch_ros2/issues/153